spec.md

```diff
@@ -2369,6 +2369,108 @@ As an example `firetruck$y$z`{.firrtl} shares a prefix with
 `firetruck$y`{.firrtl} and `firetruck`{.firrtl}, but does not share a prefix
 with `fire`{.firrtl}.
 
+# Annotations
+
+Annotations encode arbitrary metadata and associate it with zero or more
+targets in a FIRRTL circuit.
+
+Annotations are represented as a dictionary, with a "class" field which
+describes which annotation it is, and a "target" field which represents the IR
+object it is attached to. The annotation's class matches the name of a Java
+class in the Scala Chisel/FIRRTL code base. Annotations may have arbitrary
+additional fields attached. Some annotation classes extend other annotations,
+which effectively means that the subclass annotation implies to effect of the
+parent annotation.
+
+Annotations are serializable to JSON and either live in a separate file
+(e.g., during the handoff between Chisel and the SFC) or are stored in-memory
+(e.g., during SFC-based compilation). The SFC pass API requires that passes
+describe which targets in the circuit they update. SFC infrastructure then
+automatically updates annotations so they are always synchronized with their
+corresponding FIRRTL IR.
+
+An example of an annotation is the `DontTouchAnnotation`, which can be used to
+indicate to the compiler that a wire "foo" should not be optimized away.
+
+```json
+{
+  "class":"firrtl.transforms.DontTouchAnnotation",
+  "target":"~MyCircuit|MyModule>foo"
+}
+```
```

Comment:
Can you change this to not use `DontTouchAnnotation` and just use a generic annotation with a class called "Foo" or anything else. I'd prefer to not hint at optimization barrier semantics here.